scipy/_lib/_array_api.py

```diff
@@ -600,140 +601,303 @@ def is_marray(xp):
-
-    for backend in [numpy, cupy, torch, jax, dask]:
-        for attr in ['cpu', 'gpu']:
-            val = "✓" if getattr(backend, attr) else "✗"
-            val += " " * (len('{numpy.}') + len(attr) - 1)
-            setattr(backend, attr, val)
-
-    table = f"""                
-    +---------+-------------+-------------+
-    | Library | CPU         | GPU         |
-    +=========+=============+=============+
-    | NumPy   | {numpy.cpu} | n/a         |
-    +---------+-------------+-------------+
-    | CuPy    | n/a         | {cupy.gpu } |
-    +---------+-------------+-------------+
-    | PyTorch | {torch.cpu} | {torch.gpu} |
-    +---------+-------------+-------------+
-    | JAX     | {jax.cpu  } | {jax.gpu  } |
-    +---------+-------------+-------------+
-    | Dask    | {dask.cpu } | {dask.gpu } |
-    +---------+-------------+-------------+
-    """
-    return table
-
-
-def _make_capabilities_note(fun_name, capabilities):
-    table = _make_capabilities_table(capabilities)
-    note = f"""
-    `{fun_name}` has experimental support for Python Array API Standard compatible
-    backends in addition to NumPy. Please consider testing these features
-    by setting an environment variable ``SCIPY_ARRAY_API=1`` and providing
-    CuPy, PyTorch, JAX, or Dask arrays as array arguments. The following
-    combinations of backend and device (or other capability) are supported.
-    {table}
-    See :ref:`dev-arrayapi` for more information.
-    """
-    return textwrap.dedent(note)
+@dataclasses.dataclass
+class Capability:
+    display_name: str
+    # True if this backend support computation on SCIPY_DEVICE=cpu
+    cpu: bool | None
+    # True if this backend support computation on SCIPY_DEVICE=cuda
+    gpu: bool | None
+    # Reason why either cpu or gpu is not supported
+    reason: str | None = None
+
+
+@dataclasses.dataclass
+class JaxCapability(Capability):
+    # True if this function works when wrapped with jax.jit;
+    # False otherwise. Enforced by xpx.lazy_xp_function.
+    jit: bool = True
+    # Parameters for jax.jit applied by xpx.lazy_xp_function.
+    static_argnames: str | Sequence[str, ...] | None = None
+    static_argnums: int | Sequence[int, ...] | None = None
+
+
+@dataclasses.dataclass
+class DaskCapability(Capability):
+    # Maximum number of internal calls to compute() performed
+    # internally. 0 means that the function is properly lazy;
+    # 1+ means that performance will be terrible unless the user
+    # calls persist() ahead of time.
+    # Enforced by xpx.lazy_xp_function.
+    allow_compute: int = 0
+
+    # If False, chunks will be squashed into a single one, which
+    # is likely to harm performance. This typically happens when
+    # the developer implementing Array API support is unsure on
+    # whether there are any reductions along axes (a function that
+    # performs reductions may return outputs that change depending
+    # on chunk size!)
+    chunked: bool = True
```

Comment:
Use case for this will appear in https://github.com/scipy/scipy/pull/22342
I'm annoyed that it's a purely informative flag for now that needs to be kept in sync by hand. It would be far from trivial to test its correctness automatically.